### DISCUSSION TITLE
Sad about Sunsetting of Subversion support - SVN Bridge Open Source Request?

### DISCUSSION THREAD

**Discussion by schellingb** [community/community#69267] [Repositories]
### Select Topic Area

Product Feedback

### Body

First of all thank you GitHub for having support for SVN clients for over 10 years!

TortoiseSVN is my preferred tool and I have used it to contribute to all of my 69 repositories ever since I made a GitHub account many years ago. It is the workflow I like and would like to continue to use.

So with that, I'm very sad about [Subversion support getting sunset on January 8, 2024](https://github.blog/2023-01-20-sunsetting-subversion-support/). It sounds like the decision is final and there is no use of trying to appeal it (especially as a regular free user like me).

But, would GitHub be willing to release the source code of the Subversion bridge to the public?

It might enable someone like me to set up or build something like a local Subversion-to-Git proxy so I can continue contributing onto GitHub. I'm currently contemplating ways to accomplish something like this and I think getting insight into how GitHub supported SVN clients would greatly accelerate such an endeavor.

Thanks again to the people who implemented this feature and have maintained it over the many years. @schacon, @nickh and everyone else involved, you are my greatest heroes!

PS: I'm not looking for suggestions on Git GUI clients. I know that many exist (including TortoiseGit) but they are not adequate for me.

**Comment by HarshanaJayasekara** [comment]
Dear GitHub Team,

I hope this message finds you well. I want to begin by expressing my sincere gratitude to GitHub for its long-standing support of SVN clients for over a decade. As a committed user of TortoiseSVN, I've relied on this integration to contribute to my 69 repositories on GitHub ever since I created my account many years ago. It has provided me with a workflow I am comfortable with and wish to continue using.

However, I was disheartened to learn about the impending sunset of Subversion support on January 8, 2024. While I understand that this decision may be final and not open to appeal, I would like to propose an alternative solution that could benefit the GitHub community and users like me.

Would GitHub consider releasing the source code of the Subversion bridge to the public?

Such a gesture would enable users like myself to explore the possibility of setting up or building a local Subversion-to-Git proxy, allowing us to continue contributing to GitHub using our preferred workflow. I am currently exploring options to achieve this, and having access to the inner workings of GitHub's SVN client support would significantly expedite this endeavor.

I want to take this opportunity to express my gratitude to the individuals who implemented and maintained this feature over the years. To @schacon, @nickh, and all those involved, you have been my heroes in the GitHub community.

Please understand that I am not seeking alternative Git GUI client suggestions, as I am well aware that many options exist, including TortoiseGit. However, they do not align with my preferred workflow, which is why I am keen on exploring a solution that would allow me to continue using Subversion with GitHub.

Thank you for your consideration of this request, and for the outstanding support GitHub has provided to the open-source community over the years.

Sincerely,

**Comment by miqrogroove** [comment]
I just learned about SVN protocol being disabled.  I literally don't know how to contribute to Github now.  This is very sad.
